pkg/kubevirt/actuator_test.go

```diff
@@ -0,0 +1,400 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubevirt_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/openshift/cloud-credential-operator/pkg/kubevirt"
+	kubernetesErrors "k8s.io/apimachinery/pkg/api/errors"
+	"reflect"
+	"testing"
+
+	openshiftapiv1 "github.com/openshift/api/config/v1"
+	minterv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"github.com/openshift/cloud-credential-operator/pkg/operator/constants"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	kubernetesClient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	testNamespace        = "Kubevirt-cloud-credential-operator"
+	testCredRequestName  = "openshift-machine-api-kubevirt"
+	testInfrastructureName = "test-cluster-abcd"
+	testRandomSuffix     = "random"
+	testOpenshiftMachineApiKubevirtNamespace = "openshift-machine-api"
+
+)
+
+var (
+	kubevirtCredentialData = []byte("data")
+
+	kubevirtSpec = &minterv1.KubevirtProviderSpec{}
+
+	kubevirtCredentialsSecret = corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:                       constants.KubevirtCloudCredSecretName,
+			Namespace:                  constants.CloudCredSecretNamespace,
+		},
+		Data:       map[string][]byte{
+			kubevirt.KubevirtCredentialsSecretKey: kubevirtCredentialData,
+		},
+	}
+
+	kubevirtOpenshiftMmachineApiSecret = corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:                       constants.KubevirtCloudCredSecretName,
+			Namespace:                  testOpenshiftMachineApiKubevirtNamespace,
+		},
+		Data:       map[string][]byte{
+			kubevirt.KubevirtCredentialsSecretKey: kubevirtCredentialData,
+		},
+	}
+
+
+)
+
+func TestDecodeToUnknown(t *testing.T) {
+	codec, err := minterv1.NewCodec()
+	if err != nil {
+		t.Fatalf("failed to create codec %#v", err)
+	}
+	var raw *runtime.RawExtension
+	aps := minterv1.KubevirtProviderSpec{}
+	raw, err = codec.EncodeProviderSpec(&aps)
+	if err != nil {
+		t.Fatalf("failed to encode codec %#v", err)
+	}
+	unknown := runtime.Unknown{}
+	err = codec.DecodeProviderStatus(raw, &unknown)
+	if err != nil {
+		t.Fatalf("should be able to decode to Unknown %#v", err)
+	}
+	if unknown.Kind != reflect.TypeOf(minterv1.KubevirtProviderSpec{}).Name() {
+		t.Fatalf("expected decoded kind to be %s but was %s", reflect.TypeOf(minterv1.KubevirtProviderSpec{}).Name(), unknown.Kind)
+	}
+}
+
+func TestCreateCR(t *testing.T) {
+	if err := openshiftapiv1.Install(scheme.Scheme); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := minterv1.SchemeBuilder.AddToScheme(scheme.Scheme); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
fyi, there is a utility that does all the scheme setup for this operator. https://github.com/openshift/cloud-credential-operator/blob/master/pkg/util/scheme.go